app/views/carts/index.html.erb

```diff
@@ -1,12 +1,12 @@
 <h1> Cart </h1>
 
-<% @cart.contents.keys.each do |item| %>
-  <% Album.find(item.to_i) do |album| %>
-    <%= image_tag album.image_url %>
-    <%= album.title %>
-    <%= album.description %>
-    <%= album.formatted_price %>
-  <% end %>
+<% @cart.albums.each do |album| %>
+  <%= image_tag album.image_url %>
+  <%= album.title %>
+  <%= album.description %>
+  <%= album.formatted_price %>
+  <%= button_to "Remove", {:controller => :carts,
+      :action => 'destroy', :id => album.id }, :method => :delete %>
```

Comment:
Maybe instead of having this logic in the view we could create a custom route that handles this for us?